src/containers/Main/Tabs/Screens/LessonDetails/index.tsx

```diff
@@ -13,8 +13,9 @@ import {
 } from './styles';
 import { Props } from './types';
 
-export const LessonDetails: FC<Props> = () => {
+export const LessonDetails: FC<Props> = ({ route }) => {
   const { handleBack } = useConnect();
+  const { lesson } = route.params;
```

Comment:
`route` and `lesson` props should be inside useConnect. Use the useRoute hook to get the route params and extract them to this index from `useConnect` hook. Also, don't pass the entire lesson object directly on this class, you should separate the elements of the object.

in example: `const { title, author, categoryTitle, image, handleBack } = useConnect();`

Reply:
I manage to solve it in the new commit but I think you should check it to be 100% sure